pandas/conftest.py

```diff
@@ -376,12 +376,8 @@ def unique_nulls_fixture(request):
              FixedOffset(0), FixedOffset(-300), timezone.utc,
              timezone(timedelta(hours=1)),
              timezone(timedelta(hours=-1), name='foo')]
-TIMEZONE_IDS = ['None', 'UTC', 'US/Eastern', 'Asia/Tokyp',
-                'dateutil/US/Pacific', 'dateutil/Asia/Singapore',
-                'dateutil.tz.tzutz()', 'dateutil.tz.tzlocal()',
-                'pytz.FixedOffset(300)', 'pytz.FixedOffset(0)',
-                'pytz.FixedOffset(-300)', 'datetime.timezone.utc',
-                'datetime.timezone.+1', 'datetime.timezone.-1.named']
+TIMEZONE_IDS = [str(i) if not i or isinstance(i, str)
```

Comment:
doesn’t ids=repr just work on the fixture ?

Reply:
This would "work", but the result is still not acceptable, especially for tests that test tz_aware_fixture against itself (which would not even fit in one line if I use the whole screen width), which I have in #25637, for example.

For readability, the following uses the same format of [my previous comment](https://github.com/pandas-dev/pandas/pull/26600#issuecomment-497927487):
```
test_demo_tz_id['UTC'] PASSED                                                                 [  7%]
test_demo_tz_id['US/Eastern'] PASSED                                                          [ 15%]
test_demo_tz_id['Asia/Tokyo'] PASSED                                                          [ 23%]
test_demo_tz_id['dateutil/US/Pacific'] PASSED                                                 [ 30%]
test_demo_tz_id['dateutil/Asia/Singapore'] PASSED                                             [ 38%]
test_demo_tz_id[tzutc()] PASSED                                                               [ 46%]
test_demo_tz_id[tzlocal()] PASSED                                                             [ 53%]
test_demo_tz_id[pytz.FixedOffset(300)] PASSED                                                 [ 61%]
test_demo_tz_id[<UTC>] PASSED                                                                 [ 69%]
test_demo_tz_id[pytz.FixedOffset(-300)] PASSED                                                [ 76%]
test_demo_tz_id[datetime.timezone.utc] PASSED                                                 [ 84%]
test_demo_tz_id[datetime.timezone(datetime.timedelta(seconds=3600))] PASSED                   [ 92%]
test_demo_tz_id[datetime.timezone(datetime.timedelta(days=-1, seconds=82800), 'foo')] PASSED  [100%]
```

Reply:
this looks just fine

```
pandas/tests/indexes/datetimes/test_timezones.py::TestDatetimeIndexTimezones::test_tz_convert_roundtrip['UTC'] PASSED                                                                                                                                       [  7%]
pandas/tests/indexes/datetimes/test_timezones.py::TestDatetimeIndexTimezones::test_tz_convert_roundtrip['US/Eastern'] PASSED                                                                                                                                [ 15%]
pandas/tests/indexes/datetimes/test_timezones.py::TestDatetimeIndexTimezones::test_tz_convert_roundtrip['Asia/Tokyo'] PASSED                                                                                                                                [ 23%]
pandas/tests/indexes/datetimes/test_timezones.py::TestDatetimeIndexTimezones::test_tz_convert_roundtrip['dateutil/US/Pacific'] PASSED                                                                                                                       [ 30%]
pandas/tests/indexes/datetimes/test_timezones.py::TestDatetimeIndexTimezones::test_tz_convert_roundtrip['dateutil/Asia/Singapore'] PASSED                                                                                                                   [ 38%]
pandas/tests/indexes/datetimes/test_timezones.py::TestDatetimeIndexTimezones::test_tz_convert_roundtrip[tzutc()] PASSED                                                                                                                                     [ 46%]
pandas/tests/indexes/datetimes/test_timezones.py::TestDatetimeIndexTimezones::test_tz_convert_roundtrip[tzlocal()] PASSED                                                                                                                                   [ 53%]
pandas/tests/indexes/datetimes/test_timezones.py::TestDatetimeIndexTimezones::test_tz_convert_roundtrip[pytz.FixedOffset(300)] PASSED                                                                                                                       [ 61%]
pandas/tests/indexes/datetimes/test_timezones.py::TestDatetimeIndexTimezones::test_tz_convert_roundtrip[<UTC>] PASSED                                                                                                                                       [ 69%]
pandas/tests/indexes/datetimes/test_timezones.py::TestDatetimeIndexTimezones::test_tz_convert_roundtrip[pytz.FixedOffset(-300)] PASSED                                                                                                                      [ 76%]
pandas/tests/indexes/datetimes/test_timezones.py::TestDatetimeIndexTimezones::test_tz_convert_roundtrip[datetime.timezone.utc] PASSED                                                                                                                       [ 84%]
pandas/tests/indexes/datetimes/test_timezones.py::TestDatetimeIndexTimezones::test_tz_convert_roundtrip[datetime.timezone(datetime.timedelta(seconds=3600))] PASSED                                                                                         [ 92%]
pandas/tests/indexes/datetimes/test_timezones.py::TestDatetimeIndexTimezones::test_tz_convert_roundtrip[datetime.timezone(datetime.timedelta(days=-1, seconds=82800), 'foo')] PASSED                                                                        [100%]

============================================================================================================ 13 passed, 175 deselected in 0.14 seconds ============================================================================================================
(pandas) bash-3.2$ git diff
diff --git a/pandas/conftest.py b/pandas/conftest.py
index 8f71028f5..89b45f918 100644
--- a/pandas/conftest.py
+++ b/pandas/conftest.py
@@ -376,16 +376,9 @@ TIMEZONES = [None, 'UTC', 'US/Eastern', 'Asia/Tokyo', 'dateutil/US/Pacific',
              FixedOffset(0), FixedOffset(-300), timezone.utc,
              timezone(timedelta(hours=1)),
              timezone(timedelta(hours=-1), name='foo')]
-TIMEZONE_IDS = ['None', 'UTC', 'US/Eastern', 'Asia/Tokyp',
-                'dateutil/US/Pacific', 'dateutil/Asia/Singapore',
-                'dateutil.tz.tzutz()', 'dateutil.tz.tzlocal()',
-                'pytz.FixedOffset(300)', 'pytz.FixedOffset(0)',
-                'pytz.FixedOffset(-300)', 'datetime.timezone.utc',
-                'datetime.timezone.+1', 'datetime.timezone.-1.named']
 
 
-@td.parametrize_fixture_doc(str(TIMEZONE_IDS))
-@pytest.fixture(params=TIMEZONES, ids=TIMEZONE_IDS)
+@pytest.fixture(params=TIMEZONES, ids=repr)
 def tz_naive_fixture(request):
     """
     Fixture for trying timezones including default (None): {0}
@@ -393,8 +386,7 @@ def tz_naive_fixture(request):
     return request.param
 
 
-@td.parametrize_fixture_doc(str(TIMEZONE_IDS[1:]))
-@pytest.fixture(params=TIMEZONES[1:], ids=TIMEZONE_IDS[1:])
+@pytest.fixture(params=TIMEZONES[1:], ids=repr)
 def tz_aware_fixture(request):
     """
     Fixture for trying explicit timezones: {0}
```

we want the simplest things possible, having 2 lists is not great.

Reply:
@jreback : I think the list is also being used for parameterize the doc-string with `parametrize_fixture_doc`.   That being said, I'm not 100% sure why this decorator was developed, since we would mostly be reading these docstrings as developers in code, not via `help`.

Reply:
the decorator is ok i guess but it’s just more complexity i

i am harping in this because we have so much complexity already adding ids just means most maintenance

Reply:
That's fair.  I'll apply the suggested changes you made.

That being said, I would consider then removing `parmaetrize_fixture_doc` altogether (we don't use it anywhere else besides those two fixtures).

@h-vetinari : I think is a fair compromise for now between merging some of your test changes AND satisfying other developer requests.  We can always revisit later.

Reply:
yeah i think remove the decorator (here or later) is good

Reply:
@gfyoung 
I care more about the tests, that's true. The decorators predate me (and +1 for their removal), and I guess I'll be able to handle the ugly reprs... Merge away.